packages/core/src/test/amazonqFeatureDev/session/sessionStateChange.test.ts

```diff
@@ -0,0 +1,167 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import assert from 'assert'
+import sinon from 'sinon'
+import { PrepareCodeGenState, ConversationNotStartedState } from '../../../amazonqFeatureDev/session/sessionState'
+import { createMessenger } from '../utils'
+import { FeatureDevClient } from '../../../amazonqFeatureDev/client/featureDev'
+import { Session } from '../../../amazonqFeatureDev'
+import { createSessionConfig } from '../../../amazonq/commons/session/sessionConfigFactory'
+import { Uri } from 'vscode'
+
+/**
+ * Describes the test suite for the Session State Journey.
+ * This suite tests various aspects of session state transitions and error handling.
+ */
+describe('Session State Flow', () => {
+    const conversationId = 'conversation-id'
+    const uploadId = 'upload-id'
+    const tabId = 'tab-id'
+    const scheme = 'testScheme'
+    let session: Session
+    let featureDevClient: FeatureDevClient
+
+    beforeEach(async () => {
+        featureDevClient = sinon.createStubInstance(FeatureDevClient)
+        featureDevClient.createConversation = sinon.stub().resolves(conversationId)
+        featureDevClient.createUploadUrl = sinon.stub().resolves({
+            uploadId: uploadId,
+            uploadUrl: 'http://test-upload-url',
+            $response: {} as any,
+        })
+
+        const messenger = createMessenger()
+        const sessionConfig = await createSessionConfig(scheme)
+
+        session = new Session(sessionConfig, messenger, tabId, undefined, featureDevClient)
+        sinon.stub(session, 'conversationId').get(() => conversationId)
+        sinon.stub(session, 'uploadId').get(() => uploadId)
```

Comment:
why can't you set state on the `session` ? why is a stub needed? stubs and mocks should be avoided where possible. they should be a last resort, not the first thing that is tried when state needs to be set.